sindri-rs/src/custom_middleware.rs

```diff
@@ -198,13 +206,74 @@ pub fn vcr_middleware(bundle: std::path::PathBuf) -> VCRMiddleware {
     vcr
 }
 
+#[derive(Debug)]
```

Comment:
I have a stylistic preference towards including cargo docstrings - but it's not an absolute requirement here.  My rationale is that eventually I'd like to host cargo docs for the SDK, so I'm trying to be good about building those as we go.  The reasonable counter argument here is that the docs are not even generated for this submodule cause it is only public within the crate...but you never know, we could change that if we want to make these middleware defs more applicable to other things.  So that said, suggestion:
```suggestion
/// Middleware for request and response compression handling.
///
/// This implementation compresses request bodies as a stream if the body size
/// is at least `ZSTD_MIN_BODY_SIZE` and decompresses response bodies if the
/// `Content-Encoding: zstd` header is present.
#[derive(Debug)]
```